### PR TITLE
Try to XOR as many words as possible in xorbuf APIs - speeds up ChaCha20

### PR DESCRIPTION
**Pull request by julek-wolfssl**
Use xorbufout in chacha. Ran benchmarks for Chacha with a `./configure` build on my x86_64 machine.

Without this pull request:

```
$ ./wolfcrypt/benchmark/benchmark -chacha20
------------------------------------------------------------------------------
 wolfSSL version 5.5.4
------------------------------------------------------------------------------
wolfCrypt Benchmark (block bytes 1048576, min 1.0 sec each)
CHACHA                     294 MB took 1.016 seconds,  288.985 MB/s Cycles per byte =  11.77
Benchmark complete
$ ./wolfcrypt/benchmark/benchmark -chacha20
------------------------------------------------------------------------------
 wolfSSL version 5.5.4
------------------------------------------------------------------------------
wolfCrypt Benchmark (block bytes 1048576, min 1.0 sec each)
CHACHA                     278 MB took 1.017 seconds,  273.204 MB/s Cycles per byte =  12.45
Benchmark complete
$ ./wolfcrypt/benchmark/benchmark -chacha20
------------------------------------------------------------------------------
 wolfSSL version 5.5.4
------------------------------------------------------------------------------
wolfCrypt Benchmark (block bytes 1048576, min 1.0 sec each)
CHACHA                     299 MB took 1.006 seconds,  297.137 MB/s Cycles per byte =  11.44
Benchmark complete
```

With this pull request:

```
$ ./wolfcrypt/benchmark/benchmark -chacha20
------------------------------------------------------------------------------
 wolfSSL version 5.5.4
------------------------------------------------------------------------------
wolfCrypt Benchmark (block bytes 1048576, min 1.0 sec each)
CHACHA                     451 MB took 1.010 seconds,  446.210 MB/s Cycles per byte =   7.62
Benchmark complete
$ ./wolfcrypt/benchmark/benchmark -chacha20
------------------------------------------------------------------------------
 wolfSSL version 5.5.4
------------------------------------------------------------------------------
wolfCrypt Benchmark (block bytes 1048576, min 1.0 sec each)
CHACHA                     472 MB took 1.003 seconds,  470.584 MB/s Cycles per byte =   7.23
Benchmark complete
$ ./wolfcrypt/benchmark/benchmark -chacha20
------------------------------------------------------------------------------
 wolfSSL version 5.5.4
------------------------------------------------------------------------------
wolfCrypt Benchmark (block bytes 1048576, min 1.0 sec each)
CHACHA                     472 MB took 1.004 seconds,  470.026 MB/s Cycles per byte =   7.23
Benchmark complete
```

### CODE REVIEW

**Comment by julek-wolfssl** [comment]
Changes made to `XorWords` and `XorWordsOut` to accept `wolfssl_word**` so that we don't have to do pointer arithmetic after the call to these functions. Checked with godbolt.org to see if the compiler outputs different assembly.

**Comment by julek-wolfssl** [comment]
[Godbolt comparison](https://godbolt.org/#z:OYLghAFBqd5QCxAYwPYBMCmBRdBLAF1QCcAaPECAMzwBtMA7AQwFtMQByARg9KtQYEAysib0QXACx8BBAKoBnTAAUAHpwAMvAFYTStJg1DIApACYAQuYukl9ZATwDKjdAGFUtAK4sGISWakrgAyeAyYAHI%2BAEaYxCAAHKQADqgKhE4MHt6%2B/oGp6Y4CoeFRLLHxSXaYDplCBEzEBNk%2BfgG2mPZFDPWNBCWRMXGJtg1NLbntCmP9YYPlwwkAlLaoXsTI7BzmAMxhyN5YANQmO27T%2BIIAdAin2CYaAIIPj2G0c0cAbqh46EcAGiQAOokdAKADytHQEC8YQIADZJAB9AgAKiOZCOaAY0yOsMEiJR6KYpBeR3JFMpVOpNNpFPxBB2ZhRRwYSxeJgA7FYnpSGUyWXhTjznryKfxiEcIHgTjsACJHDTCo4y05uVnKoWWaxLDEmACsFiF%2BoVBvu8qOTANRoNcuFHM5dqeL2%2BvyOqhI0S8VEh0Nd6HRXqopCxAlx/vRLCYCgA1iH%2BcyCKGvIJ2c7uWT6XCBUnqUKdiK%2BXCEkSadFZQqYcWiUsg/axeTsbiGSW0UcWBWpU2ky2a1HY/XRY9KXgqFKoAzkgRiCja96Tpz1VXBFOZwQlv2Y7rzPqjukAF6YVBUZcI5HriunBUabcZhtUgD06Me72ADDYgixCBqMYURzWBBXEcyhpOk0T0EcRAAuCABKRwAO6ggoQGog%2BmbUgmLKIcQYKdmgKZJg%2Be54Iex6noS66DrSgLECCOEQlCEBSgyFGonOwboXS1IQN2eJwqxG7RnGCFIeyBacZS5bWFeIn0Uc6IHkeJ4seeYmFjSHbSRa2G4QpJFKeRqlUTS%2BGfjuMmKWRKmzsZ5Jck6Q6UhKUqqhaSoFiqsrqqZBCatJlhpsOZbWsapr6uaCosCFtqDvZHJPG8Hz%2BgCwJIREmDwYZRLopivHWWixKkveXElaV5KYUmbIOup5XZomKqxfeznSp27kWJ5aoah5WpWAFEkUqiECosQ7LapYJzhTJg2okwo29VY4npg5Lo/H8HrEEG6WZRGRxBiGvE7Zu8Z1SyPmBVyNV8YIOY0vml29m21LRLZoY4j21ZtlFi2ORSo7jqeq6zkGC5LpO06zpu25mLulnKfxqmXm5t6XZST5HC%2BeBvh%2BSbIN%2ByC/v%2BXiAcBoF4OBmCQag0FwTpyHyWhxUUjRdFgltTFZQVo3wnt/WlTxYbvQS56omx5jwkdyafkRsMc0sam85LSbmRaMv5fL95xc695SRap6tmxdbfZSmm63lH1sZujVBeSzWudempeYrfljRYgW0s9hqhRNEXttFJqxY6HArLQnD6rwfgcFopCoJwbj%2Be1ChrBsFO7DwpAEJoQcrDGID6ho%2BicJI4eZ9HnC8AoID5xnkdB6QcCwEgaAsMkdBxOQlBNy39DxMAXBmIEWCfHgmwAGp4Bl4LJIwnBpzQtAEHEFcQNEJfRGEjQAJ4z7wa/MMQG/gtE2g1NXadN9j4IMLQW816QWBRkY4i3/gxAn3gnyYBXt%2BYKoNRE1sac4SdBLu8aIxBN4eCwCXaceAWDbxWFQAwwAFBjwnlPCOs9ZAiDEOwKQMhBCKBUOoW%2BuguD6EMMYeO%2BgyYV0gCsVAU5MhfwALTgjMOXTob9nAQFcBMPwXAdhBAYOgAYZQKgSE5CkNIGQBB8IkIIgoMiGCiKGPELgkjqi1AEL0cYnhWjyI6F0OoMwVELDURomYciBGjD6KY8R6iViJ3WJsCQwdQ7F1vjHDg7oEjwmYYiI4wBkDICOH3K4ZgpS4EICQE4ZgdhcCWLwauWg5akBznnAuHAi6kAjlHLx5dK7p0zqkkOHB2E5JLvkopNdUkf2IOkZwkggA%3D%3D%3D). This shows that no extra arithmetic is necessary after the loop.

**Review by SparkiDev — DISMISSED** [review]

**Comment by bandi13** [comment]
Jenkins retest this please

**Comment by julek-wolfssl** [comment]
Used `xorbuf` API where possible. Left the math libraries and ports alone.

**Comment by bandi13** [comment]
Jenkins retest this please

**Comment by julek-wolfssl** [comment]
Jenkins retest this please

**Review by JacobBarthelmeh — DISMISSED** [review]
Please rebase and resolve the merge conflict

**Review by JacobBarthelmeh — APPROVED** [review]

File: wolfcrypt/src/misc.c [hunk 1]
```diff
@@ -256,59 +256,90 @@ WC_MISC_STATIC WC_INLINE void ByteReverseWords64(word64* out, const word64* in,
 #ifndef WOLFSSL_NO_XOR_OPS
 /* This routine performs a bitwise XOR operation of <*r> and <*a> for <n> number
 of wolfssl_words, placing the result in <*r>. */
-WC_MISC_STATIC WC_INLINE void XorWordsOut(wolfssl_word* r,
-                         const wolfssl_word* a, const wolfssl_word* b, word32 n)
+WC_MISC_STATIC WC_INLINE void XorWordsOut(wolfssl_word** r,
```
[the remark this answers is not in the document]
Comment (julek-wolfssl) [reply wolfcrypt/src/misc.c:259]: `wolfssl_word` is defined to the largest type supported by the architecture. On 64-bit systems its defined as `typedef word64 wolfssl_word;`.

**Pull request merged**